freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/user/model/UmsState.java

```diff
@@ -41,10 +43,12 @@ public UmsState(Map<User, List<Group>> userToGroupsMap, Map<String, User> adminU
 
     public UsersState getUsersState(String environmentCrn) {
 
-        //TODO: parse env and get name of env
+        // TODO: Might not be needed if customer provided groups membership is done. Then we sync all members group.
         String adminGrpName = "cpd_env_admin_ap-test-env";
+
         UsersState.Builder builder = new UsersState.Builder();
 
+        // TODO: Question: Why is this used for.
```

Comment:
this was used when groups were retrieved separately. The GetRightsResponse lists the group CRNs that a user is a member of. We needed to match the CRN to the Group.
I think we should go back to listing all the groups once. As it stands now, we are doing a lot of extra work to listGroups for every user.